VirtoCommerce.QuoteModule.Data/Services/QuoteRequestServiceImpl.cs

```diff
@@ -88,14 +90,16 @@ public IEnumerable<QuoteRequest> SaveChanges(QuoteRequest[] quoteRequests)
                     foreach (var newQuote in newQuotes)
                     {
                         // Do business logic on  quote request
-                        _eventPublisher.Publish(new QuoteRequestChangeEvent(EntryState.Added, newQuote, newQuote));
+                        changedEntries.Add(new GenericChangedEntry<QuoteRequest>(newQuote, EntryState.Added));
                         var newDbQuote = newQuote.ToDataModel(pkMap);
                         repository.Add(newDbQuote);
 
                     }
+                    _eventPublisher.Publish(new QuoteRequestChangeEvent(changedEntries));
                     repository.UnitOfWork.Commit();
                     //Copy generated id from dbEntities to model
                     pkMap.ResolvePrimaryKeys();
+                    _eventPublisher.Publish(new QuoteRequestChangeEvent(changedEntries));
```

Comment:
Why do we need to call the same `QuoteRequestChangeEvent` twice? Probably it makes sense in case of -Changing and -Changed events but do not know why we need to do it here.